python/src/aiconfig/editor/client/src/LocalEditor.tsx

```diff
@@ -29,6 +25,29 @@ export default function Editor() {
     loadConfig();
   }, [loadConfig]);
 
+  const setupTelemetryIfAllowed = useCallback(async () => {
+    const res = await ufetch.get(ROUTE_TABLE.GET_AICONFIGRC, {}); // change this to the endpoint used to get the config
+
+    const enableTelemetry = res.allow_usage_data_sharing;
+    const isDev = (process.env.NODE_ENV ?? "development") === "development";
```

Comment:
Let's do the dev check at the top so we don't need to make a request unnecessarily

Reply:
Checked for isDev before making the request, and if so return early.

Manually tested the flows and verified telemetry is only sent in prod and not dev